tasks/sep/025-mt-cannon/NestedSignFromJson.s.sol

```diff
@@ -0,0 +1,187 @@
+import {stdJson} from "forge-std/StdJson.sol";
+import {stdToml} from "forge-std/StdToml.sol";
+import {Vm, VmSafe} from "forge-std/Vm.sol";
+import {GnosisSafe} from "safe-contracts/GnosisSafe.sol";
+import {LibString} from "solady/utils/LibString.sol";
+import {Types} from "@eth-optimism-bedrock/scripts/Types.sol";
+import "@eth-optimism-bedrock/src/dispute/lib/Types.sol";
+import {DisputeGameFactory} from "@eth-optimism-bedrock/src/dispute/DisputeGameFactory.sol";
+import {FaultDisputeGame} from "@eth-optimism-bedrock/src/dispute/FaultDisputeGame.sol";
+import {PermissionedDisputeGame} from "@eth-optimism-bedrock/src/dispute/PermissionedDisputeGame.sol";
+import {SystemConfig} from "@eth-optimism-bedrock/src/L1/SystemConfig.sol";
+import {ISemver} from "@eth-optimism-bedrock/src/universal/ISemver.sol";
+
+contract NestedSignFromJson is OriginalNestedSignFromJson {
+    using LibString for string;
+
+    // Chains for this task.
+    string l1ChainName = vm.envString("L1_CHAIN_NAME");
+    string l2ChainName = vm.envString("L2_CHAIN_NAME");
+
+    // Safe contract for this task.
+    GnosisSafe ownerSafe = GnosisSafe(payable(vm.envAddress("OWNER_SAFE")));
+    GnosisSafe councilSafe = GnosisSafe(payable(vm.envAddress("COUNCIL_SAFE")));
+    GnosisSafe foundationSafe = GnosisSafe(payable(vm.envAddress("FOUNDATION_SAFE")));
+
+    // The slot used to store the livenessGuard address in GnosisSafe.
+    // See https://github.com/safe-global/safe-smart-account/blob/186a21a74b327f17fc41217a927dea7064f74604/contracts/base/GuardManager.sol#L30
+    bytes32 livenessGuardSlot = 0x4a204f620c8c5ccdca3fd54d003badd85ba500436a431f0cbda4f558c93c34c8;
+
+    SystemConfig systemConfig = SystemConfig(vm.envAddress("SYSTEM_CONFIG"));
+
+    // DisputeGameFactoryProxy address.
+    DisputeGameFactory dgfProxy;
+    // Dispute game addresses
+    address faultDisputeGameAddr;
+    address permissionedDisputeGameAddr;
+    
+    
+    // Validation expectations
+    address mips64Addr = vm.envAddress("MIPS64");
+    bytes32 immutable absolutePrestate = vm.envBytes32("ABSOLUTE_PRESTATE");
+    string expectedVmVersion = vm.envString("VM_VERSION");
+    string expectedFDGVersion = vm.envString("FDG_VERSION");
+
+    address[] extraStorageAccessAddresses;
+
+    function setUp() public {
+        string memory inputJson;
+        string memory path = "/tasks/sep/025-mt-cannon/input.json";
+        try vm.readFile(string.concat(vm.projectRoot(), path)) returns (string memory data) {
+            inputJson = data;
+        } catch {
+            revert(string.concat("Failed to read ", path));
+        }
+        
+        dgfProxy = DisputeGameFactory(systemConfig.disputeGameFactory());
+        faultDisputeGameAddr = stdJson.readAddress(inputJson, "$.transactions[0].contractInputsValues._impl");
+        permissionedDisputeGameAddr = stdJson.readAddress(inputJson, "$.transactions[1].contractInputsValues._impl");
```

Comment:
It will [revert in this case](https://github.com/mbaxter/superchain-ops/blob/2afb3ea8dde1bea743080a00c5b658288f268fcb/tasks/sep/025-mt-cannon/NestedSignFromJson.s.sol#L112).  Given this, maybe better to have a single source of truth - wdyt?

Reply:
Hmm. I think it's better to hardcode anyways because the input.json metadata fields (the `contractInputsValues`) aren't actually used to build the batch tx. So if that address is incorrect but a valid game and it deviates from the actual batch tx, then one might not detect this.

Reply:
that's a good point - updated